packages/esm-patient-registration-app/src/patient-registration/field/address/address-field.component.tsx

```diff
@@ -1,30 +1,192 @@
 
-export interface AddressFieldProps {
-  fieldDefinition: FieldDefinition;
+function parseString(xmlDockAsString: string) {
+  const parser = new DOMParser();
+  return parser.parseFromString(xmlDockAsString, 'text/xml');
+}
+function getTagAsDocument(tagName: string, template: XMLDocument) {
+  const tmp = template.getElementsByTagName(tagName)[0];
+  return tmp ? parseString(tmp.outerHTML) : parseString('');
 }
 
-export const AddressField: React.FC<AddressFieldProps> = ({ fieldDefinition }) => {
+export const AddressHierarchy: React.FC = () => {
+  const [selected, setSelected] = useState('');
+  const [addressLayout, setAddressLayout] = useState<
+    Array<{
+      id: string;
+      name: string;
+      value: string;
+      label: string;
+    }>
+  >([]);
   const { t } = useTranslation();
+  const { addressTemplate } = useContext(ResourcesContext);
+  const addressTemplateXml = addressTemplate?.results[0].value;
+  const setSelectedValue = (value: string) => {
+    setSelected(value);
+  };
+  const config = useConfig();
+  const {
+    fieldConfigurations: {
+      address: {
+        useAddressHierarchy: { enabled, useQuickSearch, searchAddressByLevel },
+      },
+    },
+  } = config;
+
+  const { setFieldValue, values, setInitialFormValues } = useContext(PatientRegistrationContext);
+  const { orderedFields, isLoadingFieldOrder, errorFetchingFieldOrder } = useOrderedAddressHierarchyLevels();
+
+  useEffect(() => {
+    const templateXmlDoc = parseString(addressTemplateXml);
+    const elementDefaults = getTagAsDocument('elementDefaults', templateXmlDoc);
+    const defaultValuesEntries = elementDefaults.getElementsByTagName('entry');
+    const defaultValues = Object.fromEntries(
+      Array.prototype.map.call(defaultValuesEntries, (entry: Element) => {
+        const [name, value] = Array.from(entry.getElementsByTagName('string'));
+        return [name.innerHTML, value.innerHTML];
+      }),
+    );
+    const nameMappings = getTagAsDocument('nameMappings', templateXmlDoc);
+    const properties =
+      Array.from(nameMappings.getElementsByTagName('property')).length > 0
+        ? nameMappings.getElementsByTagName('property')
+        : nameMappings.getElementsByTagName('entry');
+
+    const propertiesObj = Array.prototype.map.call(properties, (property: Element) => {
+      const name = property.getAttribute('name') ?? property.getElementsByTagName('string')[0].innerHTML;
+      const label = property.getAttribute('value') ?? property.getElementsByTagName('string')[1].innerHTML;
+      /*
+        DO NOT REMOVE THIS COMMENT UNLESS YOU UNDERSTAND WHY IT IS HERE
+
+        t('postalCode', 'Postal code')
+        t('address1', 'Address line 1')
+        t('address2', 'Address line 2')
+        t('countyDistrict', 'District')
+        t('stateProvince', 'State')
+        t('cityVillage', 'city')
+        t('country', 'Country')
+        t('countyDistrict', 'District')
+      */
```

Comment:
Why is this comment here? :)

Reply:
I wrote this comment a while ago when a similar one was deleted.

Basically, we want to have the `en.json` file to have translations for the fields that are part of the RefAPp, but the list of address properties is dynamic, so, we don't just have the usual `t('postalCode', 'Postal Code');`. Instead, we have what happens a few lines down where `labelText` is set to `t(attributes.label)`. Our `extract-translations` script isn't smart enough to know what the possible values passed to that function are, so this comment provides that script with the list.

If this comment is removed, the `extract-translations` script will remove these keys from the `en.json` file.

Reply:
I think maybe @mogoodrich is implying that this explanation should be more explicit in the code comments, rather than here on the PR discussion :)

Reply:
I got that. I was trying to point out that this wasn't new per-se, just in a new location. But, yes, it would be better to have some standardised text for this.

Reply:
Well, I was really both noting that it should be more explicit and also wanted to understand it myself... :)

Are you saying that during build any translation that can't be explicity found in the found the code would be automatically removed from the en.json file?  (I feel like this is something I'm almost certainly during in the Dispensing ESM for various status methods and so would need to fix).

Reply:
Yes @mogoodrich the entries will be deleted from en.json

Reply:
Technically, it’s [optional](https://github.com/openmrs/openmrs-esm-patient-management/blob/399a70ec5a96c25b52fb18704f333c26d160d106/tools/i18next-parser.config.js#L18), but yes, we prune keys that aren’t used.

The code that does this is executed as part of a Git commit hook, so it’s basically done on commit; this gives you an opportunity to see the changes before anything is actually done.